apiml-common/src/main/java/org/zowe/apiml/product/eureka/client/ApimlPeerEurekaNode.java

```diff
@@ -372,11 +373,39 @@ public static class ReplicationTaskProcessor implements TaskProcessor<Replicatio
 
         private static final Pattern READ_TIME_OUT_PATTERN = Pattern.compile(".*read.*time.*out.*");
 
+        private final NetworkIssueCounter networkIssueCounter = new NetworkIssueCounter();
+
         public ReplicationTaskProcessor(String peerId, HttpReplicationClient replicationClient) {
             this.replicationClient = replicationClient;
             this.peerId = peerId;
         }
 
+        static class NetworkIssueCounter {
+
+            static final int MAX_RETRIES = 10;
+            final AtomicInteger counter = new AtomicInteger(0);
+
+            public void success() {
+                int count = counter.get();
+                if (count > 0) {
+                    log.trace("Network error indicator was reset. The number of errors was {}/{}", count, MAX_RETRIES);
+                }
+                counter.set(0);
+            }
+
+            public void fail(String errorMessage) {
+                int count = counter.getAndUpdate(prev -> Math.min(prev + 1, MAX_RETRIES));
+                log.trace("Network error ({}) occurred. The number of errors is {}{}/{}. The network error status is considered as {}.",
+                    errorMessage, counter.get(), count >= MAX_RETRIES ? "+" : "",
+                    MAX_RETRIES, isPermanent() ? "permanent" : "temporary");
+            }
+
+            public boolean isPermanent() {
```

Comment:
How about renaming this method to something like `hasReachedMax` or similar, at least in my first read I assumed permanent meant it wasn't going to try again, but if I understand it right, once it has a successful attempt it restarts the counter, so it's not permanent really. 
Also, I think this looks like a circuit breaker use case, have you considered using hystrix for example? it would allow to configure features like exponential backoff, I think it's alread in the classpath

Reply:
It should mean that the current network issue is considered permanent. This can change in the future when the network recovers. But I don't mind renaming it if you find it better.

I don't have any experience with hystrix. I don't mind using it but you would probably have to show me how.

Reply:
I saw the rename and I'm ok with it, at least I think it's more clear when reading it.
As for the second topic, I think it can be left for a future enhancement, I think some of the features of this class resemble a cb use case, but I need to look into it in more detail really.

Reply:
I would be very interested in it to learn something new. But probably not now :-)